netket/operator/_local_operator.py

```diff
@@ -540,15 +543,21 @@ def _add_operator(self, operator: Array, acting_on: List[int]):
         # If overlapping support, add the local operators themselves
         if support_i is not None:
             dim = min(operator.shape[0], self._operators[support_i].shape[0])
-            _opv = self._operators[support_i][:dim, :dim]
-            _opv += operator[:dim, :dim]
+            self._operators[support_i][:dim, :dim] += operator[:dim, :dim]
 
             n_local_states_per_site = np.asarray(
                 [self.hilbert.size_at_index(i) for i in acting_on]
             )
-
+            assert (
+                self._operators[support_i].shape[0]
+                == self._operators[support_i].shape[1]
+            )
+            _op_coo = self._operators[support_i].tocoo()
```

Comment:
I think idealy we would use the csr structure inside of `_append_matrix`. 
The conversion to coo is slow, and useless...

Can you adapt the code in `_append_matrix` to work with the csr?

Reply:
else just use coo from the beginning.
though i'd prefer csr

Reply:
Ok, I didn't notice you could access the rows and column indices using `.nonzero()`. I will change this.